hail/python/hail/linalg/blockmatrix.py

```diff
@@ -1427,6 +1459,47 @@ def __matmul__(self, b):
 
         return BlockMatrix(BlockMatrixDot(self._bmir, b._bmir))
 
+    @typecheck_method(b=oneof(np.ndarray, block_matrix_type), split_on_inner=int, path_prefix=str)
+    def tree_matmul(self, b, split_on_inner, path_prefix):
```

Comment:
I think adding a default argument for path_prefix of `None` and creating a temp file in that case will improve usability.